src/theory/strings/rewrites

```diff
@@ -247,11 +216,6 @@
   (str.contains (str.++ x y z) w)
   (or (str.contains x w) (str.contains (str.++ y z) w)))
 
-(define-cond-rule str-contains-lt-len ((x ?Seq) (y ?Seq))
```

Comment:
This needs an additional application of `evaluation` to be subsumed since the new rule has `(= x y)` has a target. How would this work if the arguments are symbolic and we only know the length?

Reply:
The other RARE rule needed is `str-eq-len-false`, I'll add this to the description of the PR.